src/Interpreters/DatabaseCatalog.h

```diff
@@ -165,10 +165,13 @@ class DatabaseCatalog : boost::noncopyable
     void updateDependency(const StorageID & old_from, const StorageID & old_where,const StorageID & new_from, const StorageID & new_where);
 
     /// If table has UUID, addUUIDMapping(...) must be called when table attached to some database
-    /// and removeUUIDMapping(...) must be called when it detached.
+    /// removeUUIDMapping(...) must be called when it detached,
+    /// and removeUUIDMappingFinally(...) must be called when table is dropped and its data removed from disk.
+    /// To "lock" some UUID and prevent collision, addUUIDMapping(...) may be called with nullptr arguments.
```

Comment:
Which collision do we want to prevent? Seems like we are "booking" ID? Add examples in the comments.